### PR TITLE
Add new person fields

### PR DESCRIPTION
**Pull request by houtenteo**
Added the optional fields Address, Gender, Age, Interests

### CODE REVIEW

**Review by Kangxun — COMMENTED** [review]
Minor clarifications otherwise LGTM

File: src/main/java/seedu/address/model/person/Gender.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    /**
+     * Returns true if a given string is a valid email.
+     */
+    public static Boolean isValidGender(String gender) {
+        if (gender == null) {
+            return true;
+        }
+        String test = gender.toUpperCase(Locale.ROOT);
+        //System.out.println(test);
+        return test.equals("M") || test.equals("F") || test.equals("N.A");
```
[the remark this answers is not in the document]
Comment (houtenteo) [reply src/main/java/seedu/address/model/person/Gender.java]: Oh u can use regex like that? HAHA didn't know. OK ill play ard with it

File: src/main/java/seedu/address/model/person/interests/Interest.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.util.Objects;
+
+public class Interest {
+    public static final String MESSAGE_CONSTRAINTS = "Interests should be alphanumeric";
+    public static final String NO_INTEREST = "N.A";
+
+    /*
+     * The first character of the Interest must not be a whitespace,
```
[the remark this answers is not in the document]
Comment (houtenteo) [reply src/main/java/seedu/address/model/person/interests/Interest.java]: Ah you right. I was testing out different regex but forgot to change the constraints. Nice spot

File: src/main/java/seedu/address/model/person/Address.java [hunk 3]
```diff
@@ -25,10 +25,10 @@
      * @param address A valid address.
      */
     public Address(String address) {
+        checkArgument(isValidAddress(address), MESSAGE_CONSTRAINTS);
```
[the remark this answers is not in the document]
Comment (Kangxun) [reply src/main/java/seedu/address/model/person/Address.java:30]: It was in a if-else block before moving it up, but with the new code (from the next few commits) it looks alright.

File: src/main/java/seedu/address/model/person/Age.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
+package seedu.address.model.person;
+
+import java.util.Objects;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+public class Age {
+    public static final String MESSAGE_CONSTRAINTS = "Age should only contain numbers";
```
[the remark this answers is not in the document]
Comment (Kangxun) [reply src/main/java/seedu/address/model/person/Age.java]: Good point, will just have to take note in future

**Pull request merged**